ocs_ci/ocs/longevity.py

```diff
@@ -618,6 +656,120 @@ def stage_2(
             )
             time.sleep(delay)
 
+    def stage3(
+        self,
+        project_factory,
+        num_of_pvc,
+        num_of_obc,
+        pvc_size=None,
+        delay=60,
+        run_time=1440,
+    ):
+        """
+        Concurrent bulk operations of following
+            PVC creation - all supported types  (RBD, CephFS, RBD-block)
+            PVC deletion - all supported types  (RBD, CephFS, RBD-block)
+            OBC creation
+            OBC deletion
+            APP pod creation - all supported types  (RBD, CephFS, RBD-block)
+            APP pod deletion - all supported types  (RBD, CephFS, RBD-block)
+
+        Args:
+            project_factory : Fixture to create a new Project.
+            num_of_pvc (int): Bulk PVC count
+            num_of_obc (int): Bulk OBC count
+            pvc_size (str): size of all pvcs to be created with Gi suffix (e.g. 10Gi).
+            If None, random size pvc will be created
+            delay (int): Delay in seconds before starting the next cycle
+            run_time (int): The amount of time the particular stage has to run (in minutes)
+
+        """
+        end_time = datetime.now() + timedelta(minutes=run_time)
+        cycle_count = 1
+        while datetime.now() < end_time:
+            log.info(f"Current time is {datetime.now()}")
+            log.info(f"End time is {end_time}")
+            log.info(
+                f"##############[STARTING CYCLE:{cycle_count}]####################"
+            )
+            namespace = f"{STAGE_3_PREFIX}{cycle_count}"
+            project_factory(project_name=namespace)
+            log.info(
+                "Creating the initial resources required for PVC/OBC/POD deletion operations concurrently"
+            )
+            resource_to_delete = [
+                ThreadPoolExecutor(max_workers=1).submit(
+                    self.create_stagebuilder_all_pvc_types,
+                    num_of_pvc,
+                    namespace,
+                    pvc_size,
+                    kube_job_name="delete_all_pvc_job_profile",
+                ),
+                ThreadPoolExecutor(max_workers=1).submit(
+                    self.create_stagebuilder_obc,
+                    num_of_obc,
+                    namespace,
+                    obc_kube_job_name="delete_obc_job_profile",
+                ),
+                ThreadPoolExecutor(max_workers=1).submit(
+                    self.create_stagebuilder_pods_with_all_pvc_types,
+                    num_of_pvc,
+                    namespace,
+                    pvc_size,
+                    pvc_kube_job_name="delete_all_pvc_for_pod_attach_job_profile",
+                    pod_kube_job_name="delete_all_pods_job_profile",
+                ),
+            ]
+            resource_to_delete_job_file = []
+            # waiting for the resource thread to complete
+            for resource in resource_to_delete:
+                log.info("Waiting for the resource thread to complete")
```

Comment:
Please move this line to be before the for-loop.
It will just print:
Waiting for the resource thread to complete
Waiting for the resource thread to complete
Waiting for the resource thread to complete

Reply:
Done